src/librustc_typeck/diagnostics.rs

```diff
@@ -1130,6 +1244,96 @@ impl Bytes { ... } // error, same as above
 ```
 "##,
 
+E0117: r##"
+You got this error because because you tried to implement a foreign
+trait on a foreign (with maybe a foreign type parameter). Erroneous
```

Comment:
I think its fine, but you need to fix your sentence anyway. First of all foreign type parameter note is in the wrong location (should go after “foreign trait”), you implement traits **for** types, not **on** types and current sentence ends abruptly (should be “foreign type” not “foreign”).